### DISCUSSION TITLE
Comparing two column values in supabase-js query

### DISCUSSION THREAD

**Discussion by nthouliss** [supabase/supabase#6678] [Questions] [answered]
I'm trying to get the results in a table by checking that a value in one column is less than that of a value in another column.

For instance the Postgres query would be something like this:

`
SELECT example_limit, current_total
FROM example
WHERE current_total < example_limit;
`

I've tried to use queries like these in supabase js:

`
const { data, error } = await supabase.from("example").lt("current_total", "example_limit");
`

`
const { data, error } = await supabase.from("example").filter("current_total", "lt", "example_limit");
`

However, I end up with casting errors and have only managed to get it working with using a numerical value. Is it possible to compare two columns from the same table using supabase-js?

It isn't obvious if this is something that's achievable or not when reading through the docs and I guess I'd like to know if I'll have to use a workaround (like an rpc).

**Comment by steve-chavez** [comment, marked as answer]
Not directly bu you can create a computed column for that - it's less code than a full set returning function:

Check https://github.com/supabase/supabase/discussions/6005#discussioncomment-2416130

**Reply by nthouliss** [reply]
Cheers! Was looking everywhere to see if this had been asked before, guess my search was off. Much appreciated :)
